client/state/jetpack/connection/reducer.js

```diff
@@ -45,6 +49,32 @@ export const requests = createReducer( {}, {
 	[ JETPACK_CONNECTION_STATUS_REQUEST_SUCCESS ]: createRequestReducer( false )
 } );
 
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack user connection data
+ *
+ * @param  {Array}  state  Current state
+ * @param  {Object} action action
+ * @return {Array}         Updated state
+ */
+export const dataItems = createReducer( {}, {
+	[ JETPACK_USER_CONNECTION_DATA_RECEIVE ]: ( state, { siteId, data } ) => Object.assign( {}, state, { [ siteId ]: data } )
+} );
+
+/**
+ * `Reducer` function which handles request/response actions
+ * concerning Jetpack user connection data requests
+ *
+ * @param {Object} state - current state
+ * @param {Object} action - action
+ * @return {Object} updated state
+ */
+export const dataRequests = createReducer( {}, {
+	[ JETPACK_USER_CONNECTION_DATA_REQUEST ]: createRequestReducer( true ),
```

Comment:
Thanks, I always forget about this one! Addressed in https://github.com/Automattic/wp-calypso/pull/15599/commits/6e87af6df408064765d44afdf4b8196ead451e7f.